src/Command/RunCommand.php

```diff
@@ -85,28 +95,34 @@ protected function doExecute(): int
         // TODO: Parallelize tasks
         // Execute task list
         $exitCode = 0;
-        $keepGoing = $this->input->getOption('keep-going');
+        $fastFinish = $this->input->getOption('fast-finish');
         foreach ($taskList->getIterator() as $task) {
             $taskOutput = new BufferedOutput($consoleOutput);
             try {
                 $task->run($taskOutput);
-            } catch (RuntimeException $throwable) {
+            } catch (PluginApiRuntimeException $throwable) {
                 $taskOutput->writeln(
                     $throwable->getMessage(),
                     BufferedOutput::VERBOSITY_NORMAL,
-                    BufferedOutput::CHANNEL_STRERR
+                    BufferedOutput::CHANNEL_STDERR
                 );
                 $exitCode = (int) $throwable->getCode();
                 $exitCode = $exitCode === 0 ? 1 : $exitCode;
 
-                if (!$keepGoing) {
+                if ($fastFinish) {
                     $taskOutput->release();
-                    return $exitCode;
+                    break;
                 }
             }
             $taskOutput->release();
         }
 
+        $report->complete($exitCode === 0 ? Report::STATUS_PASSED : Report::STATUS_FAILED);
+        ReportWriter::writeReport(getcwd() . '/' . $projectConfig->getArtifactOutputPath(), $report);
+        CheckstyleReportWriter::writeReport(getcwd() . '/' . $projectConfig->getArtifactOutputPath(), $report);
+        ConsoleWriter::writeReport($this->output, $report);
```

Comment:
Which report is used should be configurable by the command options but my be handled in a separate PR.